src/lib/fountain/fountain_decoder_sink.h

```diff
@@ -78,7 +78,7 @@ class fountain_decoder_sink
 			return false;
 
 		// find or create
-		auto p = _streams.emplace(std::piecewise_construct, std::make_tuple(md.id()), std::make_tuple(md.file_size(), _chunkSize));
+		auto p = _streams.try_emplace(stream_slot(md), md.file_size(), _chunkSize);
```

Comment:
`try_emplace` does what I actually want, which is to not construct the element unless it's new.